Allow users to dynamically add new files and remove existing ones from the tailer

Kami opened the pull request:
With this change user can now dynamically add new files and remove existing ones from the tailer during run time using `add_file` and `remove_file` methods respectively.

An alternative which is less flexible would be to instantiate new tailer instance every time user wants either to add or remove a monitored file.

We use this functionality inside our file watch StackStorm sensor (https://github.com/StackStorm/st2/blob/1a020ccceebf77c6c6c1ef2e07f55dee82733bd0/contrib/linux/sensors/file_watch_sensor.py#L35). We allow users to update a list of files which are monitored during the run time.

TODO:

- [x] Add tests

kvdveer commented:
Thanks for the patch. On first inspection, it looks good. I'll merge it when tests catch up.

Kami commented:
I've fixed an existing failing test and pep8 violations (3b7760cac1175f6490dc192c6c5301ec1de40693. 0d3159bf20d696016e15eea7c84741272763ada9) and added a new test case for add_file and remove_file (6689708303e2ad748ee0cdc64ca976841e644b80).

Tests fail locally under Python 3.4, but the failure appears to be unrelated to any of my changes - https://gist.github.com/Kami/a297d7502f1a901bd923.
